libraries/botbuilder-dialogs/src/skillDialog.ts

```diff
@@ -162,35 +177,18 @@ export class SkillDialog extends Dialog {
             throw new TypeError(`"activity" is undefined or null in options.`);
         }
 
-        // Only accept Message or Event activities
-        if (dialogArgs.activity.type !== ActivityTypes.Message && dialogArgs.activity.type !== ActivityTypes.Event) {
-            // Just forward to the remote skill
-            throw new TypeError(`Only "${ ActivityTypes.Message }" and "${ ActivityTypes.Event }" activities are supported. Received activity of type "${ dialogArgs.activity.type }" in options.`);
-        }
-
         return dialogArgs;
     }
 
     private async sendToSkill(context: TurnContext, activity: Activity, connectionName: string): Promise<Activity> {
-        // Create a conversationId to interact with the skill and send the activity
-        const conversationIdFactoryOptions: SkillConversationIdFactoryOptions = {
-            fromBotOAuthScope: context.turnState.get(context.adapter.OAuthScopeKey),
-            fromBotId: this.dialogOptions.botId,
-            activity: activity,
-            botFrameworkSkill: this.dialogOptions.skill
-        };
-
-        // Create a conversationId to interact with the skill and send the activity
-        let skillConversationId: string;
-        try {
-            skillConversationId = await this.dialogOptions.conversationIdFactory.createSkillConversationIdWithOptions(conversationIdFactoryOptions);
-        } catch (err) {
-            if (err.message !== 'Not Implemented') throw err;
-            // If the SkillConversationIdFactoryBase implementation doesn't support createSkillConversationIdWithOptions(),
-            // use createSkillConversationId() instead.
-            skillConversationId = await this.dialogOptions.conversationIdFactory.createSkillConversationId(TurnContext.getConversationReference(activity) as ConversationReference);
+        if (activity.type === ActivityTypes.Invoke) {
```

Comment:
having to switch expectReplies on for an invoke seems a little odd.

Reply:
if the dialog is interacting with a skill that only works with invokes, this makes sure that EoC replies are buffered with the invokeResponse and the dialog ends when an EoC comes back.